scalafix/rules/src/main/scala/utils/CompilerService.scala

```diff
@@ -38,15 +39,43 @@ object CompilerService {
     }
   }
 
-  def getContext(name: Token, g: Global)(implicit unit: g.CompilationUnit): Option[g.Context] =
-    getContext(name.pos.start, g)
+  def getContext(name: Term, g: Global)(implicit unit: g.CompilationUnit): Option[g.Context] = {
+    val gpos = unit.position(name.pos.start)
+    getContext(gpos, g)
+  }
+
+  def getContext(tree: Tree, g: Global)(implicit unit: g.CompilationUnit): Option[g.Context] = {
+    val gpos = getExactPos(tree.pos, g)
+    getContext(gpos, g)
+  }
 
-  def getContext(name: Term.Name, g: Global)(implicit unit: g.CompilationUnit): Option[g.Context] =
-    getContext(name.pos.start, g)
+  def getGlobalTree(tree: Tree, g: Global)(implicit unit: g.CompilationUnit): Option[g.Tree] = {
```

Comment:
In another commit, I will ad comments, as soon as I am completely sure of the stategy to ue to locate correctly and precisely the tree.